### PR TITLE
Initialise Spark Agent

### PR DESCRIPTION
**Pull request by ulixius9**

### CODE REVIEW

**Comment by harshach** [comment]
@ulixius9 we should add APACHE License content as well

**Comment by ulixius9** [comment]
@harshach I've already added a apache APACHE License [here](https://github.com/open-metadata/openmetadata-spark-agent/blob/main/LICENSE)

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+      </developer>
+  </developers>
+   <scm>
+    <connection>scm:git:git@github.com:Open-Metadata/OpenMetadata</connection>
+    <developerConnection>scm:git:git@github.com:Open-Metadata/OpenMetadata</developerConnection>
+    <url>scm:git:git@github.com:Open-Metadata/OpenMetadata</url>
+  </scm>
+   <properties>
+      <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+      <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (harshach) [inline pom.xml]: should this java 17? not sure what spark's java version it supports

File: src/main/java/org/openmetadata/transport/OpenMetadataConfig.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+ /*
+  * This code has been referenced from 
+  * https://github.com/Natural-Intelligence/openLineage-openMetadata-transporter.git
+  */
+  
+package org.openmetadata.transport;
+
+import io.openlineage.client.transports.TransportConfig;
+import lombok.*;
```
Comment (harshach) [inline src/main/java/org/openmetadata/transport/OpenMetadataConfig.java]: avoid wild-card imports

File: src/main/java/org/openmetadata/transport/OpenMetadataTransport.java [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.transports.Transport;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.*;
```
Comment (harshach) [inline src/main/java/org/openmetadata/transport/OpenMetadataTransport.java]: wild-card import

**Comment by harshach** [comment]
@ulixius9 also add spotless for the code styling https://github.com/open-metadata/OpenMetadata/blob/8f37d7da05225187687d2eaf547884f712e18ac4/pom.xml#L757 and post adding that , run ```mvn spotless:apply```

File: pom.xml (hunk 1 above)
Comment (ulixius9) [reply pom.xml]: 17 is not supported but 11 is, so have changed it to 11

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**